src/raw/cgroupv2.go

```diff
@@ -0,0 +1,156 @@
+// Package raw fetches raw system-level metrics as they are presented by the operating system
+package raw
+
+import (
+	"errors"
+	"path/filepath"
+	"strconv"
+	"time"
+
+	cgroupsV2 "github.com/containerd/cgroups/v2"
+	cgroupstatsV2 "github.com/containerd/cgroups/v2/stats"
+	"github.com/docker/docker/api/types"
+	"github.com/newrelic/infra-integrations-sdk/log"
+)
+
+// CgroupsV2Fetcher fetches the metrics that can be found in cgroups (v2) file system
+type CgroupsV2Fetcher struct {
+	hostRoot        string
+	systemCPUReader SystemCPUReader
+}
+
+// NewCgroupsV2Fetcher creates a new cgroups data fetcher.
+func NewCgroupsV2Fetcher(hostRoot string, systemCPUReader SystemCPUReader) (*CgroupsV2Fetcher, error) {
+	return &CgroupsV2Fetcher{
+		hostRoot:        hostRoot,
+		systemCPUReader: systemCPUReader,
+	}, nil
+}
+
+// Fetch get the metrics that can be found in cgroups v2 file system
+// Unlike v1, cgroup v2 has only single hierarchy.
+//TODO: populate also network from libcgroups
+func (cg *CgroupsV2Fetcher) Fetch(c types.ContainerJSON) (Metrics, error) {
+	stats := Metrics{}
+
+	pid := c.State.Pid
+	containerID := c.ID
+	driver := c.Driver
+
+	var (
+		cgroupInfo *cgroupV2Paths
+		err        error
+	)
+
+	cgroupInfo, err = getCgroupV2Paths(cg.hostRoot, pid, driver, containerID)
+	if err != nil {
+		return stats, err
+	}
+
+	manager, err := cgroupsV2.LoadManager(cgroupInfo.MountPoint, cgroupInfo.Group)
+	if err != nil {
+		return stats, err
+	}
+
+	metrics, err := manager.Stat()
+	if err != nil {
+		return stats, err
+	}
+
+	stats.Time = time.Now()
+
+	if stats.Pids, err = cg.pids(metrics); err != nil {
+		log.Error("couldn't read pids stats: %v", err)
+	}
+
+	if stats.CPU, err = cg.cpu(metrics); err != nil {
+		log.Error("couldn't read cpu stats: %v", err)
+	}
+
+	// TODO: missing metric
+	//cgroupFullPathBlkio, err := cgroupInfo.getFullPath(cgroups.Blkio)
+	//
+	//if err == nil {
+	//	if stats.Blkio, err = blkio(cgroupFullPathBlkio); err != nil {
+	//		log.Error("couldn't read blkio stats: %v", err)
+	//	}
+	//} else {
+	//	log.Error("couldn't read blkio stats: %v", err)
+	//}
```

Comment:
Nit: the _TODO_ and commented code are remainders of the blkio missing metric. As you has already made it explicit in the PR description, maybe we can get rid of this comment. WDYT?

Reply:
Sure 👍